float64 velocity
float64 delta
